programs/zk-token-proof-tests/tests/process_transaction.rs

```diff
@@ -1065,15 +1065,14 @@ async fn test_verify_proof_without_context<T, U>(
 
     let client = &mut context.banks_client;
     let payer = &context.payer;
-    let recent_blockhash = context.last_blockhash;
 
     // verify a valid proof (wihtout creating a context account)
     let instructions = vec![proof_instruction.encode_verify_proof(None, success_proof_data)];
     let transaction = Transaction::new_signed_with_payer(
         &instructions.with_max_compute_unit_limit(),
         Some(&payer.pubkey()),
         &[payer],
-        recent_blockhash,
+        client.get_latest_blockhash().await.unwrap(),
```

Comment:
I also wonder why a migration to a newer version of `dalek` would cause a change in the way the latest blockhash is read in the tests.
If there is a bug in the way these tests are written, it should be a separate PR.